core/templates/pages/admin-page/navbar/admin-navbar.directive.spec.ts

```diff
@@ -0,0 +1,215 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for admin navbar component.
+ */
+
+import { importAllAngularServices } from 'tests/unit-test-utils';
+
+describe('Admin Navbar component', () => {
+  var $rootScope = null;
+  var ctrl = null;
+  var $scope = null;
+  var $q = null;
+  var UserService = null;
+  var AdminRouterService = null;
+  var userProfileImage = 'profile-data-url';
+  var userInfo = {
+    isModerator: () => true,
+    getUsername: () => 'username1',
+    isSuperAdmin: () => true
+  };
+  var imagePath = '/path/to/image.png';
+  var profileUrl = '/profile/username1';
+
+  importAllAngularServices();
+
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.inject(function($injector, $componentController) {
+    UserService = $injector.get('UserService');
+    AdminRouterService = $injector.get('AdminRouterService');
+    $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+    $q = $injector.get('$q');
+    ctrl = $componentController('adminNavbar', {
+      $scope: $scope
+    });
+  }));
+
+  beforeEach(() => {
+    spyOn(UserService, 'getProfileImageDataUrlAsync')
+      .and.returnValue($q.resolve(userProfileImage));
+    spyOn(UserService, 'getUserInfoAsync').and.returnValue(
+      $q.resolve(userInfo));
+    ctrl.$onInit();
+    $rootScope.$apply();
+  });
+
+  it('should initialize ctrl properties after directive is initialized', () => {
+    expect(ctrl.profilePictureDataUrl).toBe(userProfileImage);
+    expect(ctrl.getStaticImageUrl(imagePath)).toBe(
+      '/assets/images/path/to/image.png');
+    expect(ctrl.username).toBe('username1');
+    expect(ctrl.isModerator).toBe(true);
+    expect(ctrl.isSuperAdmin).toBe(true);
+    expect(ctrl.profileUrl).toEqual(profileUrl);
+    expect(ctrl.logoutUrl).toEqual('/logout');
+    expect(ctrl.profileDropdownIsActive).toBe(false);
+    expect(ctrl.dropdownMenuisActive).toBe(false);
+  });
+
+  it('should be call showtab in AdminRouterService', () => {
```

Comment:
Not sure what this means.

Reply:
There is a function called showtab in AdminRouterService. So the admin navbar should call that function.

https://github.com/oppia/oppia/blob/674740a3c732d0457300a36f82c11cf7a575f6b9/core/templates/pages/admin-page/services/admin-router.service.ts#L53-L57

Reply:
Yep I understood what it does but the description of the test looks a bit off.

Reply:
`Should call showTab function in AdminRouterService when showTab in AdminManbar is called`

How does this sound?

Reply:
Maybe you meant
```suggestion
  it('should call showTab in AdminRouterService', () => {
```

Reply:
I thought it might be better to specify the function and the file, since they have the same function name?

Reply:
Makes sense but please pay attention to the test description next time.

Also, what actually happens when AdminRouterService.showTab() is called? Does it change the current tab?

Reply:
>Makes sense but please pay attention to the test description next time.
Sure!
> Also, what actually happens when AdminRouterService.showTab() is called? Does it change the current tab?
No, it seems to just assign the `tabhash` to the `currentTabHash` variable

Reply:
Yep but what happens when tabHash is empty (i.e. when you do a `AdminRouterService.showTab()`)?

Reply:
Maybe activities tab is selected since that is the default?

Reply:
yes it is default,

https://github.com/oppia/oppia/blob/674740a3c732d0457300a36f82c11cf7a575f6b9/core/templates/pages/admin-page/services/admin-router.service.ts#L29-L30

Reply:
So in that case, can we modify the test such that we expect the current tab to be "Activities" and change the description accordingly?

Reply:
DONE